### ISSUE TITLE
Implement build_mlp and train_model (PyTorch)

### ISSUE THREAD

**Issue by kakhidr** [kakhidr/XAI_Drift_Detection#2]
### Required functions
**File:** src/models/mlp.py
```
def build_mlp(cfg, input_dim: int):
    ...

def train_model(model, X_train, y_train, cfg):
    ...

```
**Requirements**
- Output logits [N, num_classes]
- Use config:
   - hidden_dims, dropout, epochs, batch_size, lr, num_classes
- Use CrossEntropyLoss + Adam
- Support CPU and GPU (cfg["run"]["use_cuda"])

**Acceptance criteria**

- Training runs without crashing on both datasets
- Model can perform inference on test data
- Seed makes results reproducible (same run outputs stable)
